packages/client-sdk-nodejs/src/auth-client.ts

```diff
@@ -0,0 +1,57 @@
+import {AuthClient as InternalAuthClient} from './internal/auth-client';
+import {IAuthClient} from '@gomomento/sdk-core/dist/src/internal/clients/auth/IAuthClient';
+import {AbstractAuthClient} from '@gomomento/sdk-core/dist/src/internal/clients/auth/AbstractAuthClient';
+import {MomentoLogger, RefreshApiToken} from '.';
+import {AuthClientProps} from './auth-client-props';
+import {GenerateApiToken} from '@gomomento/sdk-core';
+
+export class AuthClient extends AbstractAuthClient implements IAuthClient {
+  private readonly logger: MomentoLogger;
+
+  constructor(props: AuthClientProps) {
+    const authClient = new InternalAuthClient({
+      credentialProvider: props.credentialProvider,
+      configuration: props.configuration,
+    });
+
+    super({createAuthClient: () => authClient});
+
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    this.logger.info('Creating Momento AuthClient');
+  }
+
+  /**
+   * Gets a api token, refresh token given a valid session token.
+   *
+   * @param {string} sessionToken - The session token to allow access for generation of api tokens.
+   * @param {string} validForSeconds - How long the token is valid for in seconds.
+   * @returns {Promise<GenerateApiToken.Response>} -
+   * {@link GenerateApiToken.Success} containing the api token, refresh token, origin an how long the token is valid for.
+   * {@link GenerateApiToken.Error} on failure.
+   */
+  public async generateApiToken(
+    sessionToken: string,
+    validForSeconds?: number
```

Comment:
this is another thing that I'm not sure I want in our public API, see: 

https://github.com/momentohq/client-sdk-nodejs/issues/394